lib/bob/job/os_versions.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Bob.Job.OSVersions do
+  @max_alpine_versions 2
+
+  # TODO: Automate picking ubuntu and debian os versions
+
+  @static_os_versions %{
+    "ubuntu" => [
+      "groovy-20210325",
+      "focal-20210325",
+      "bionic-20210325",
+      "xenial-20210114",
+      "trusty-20191217"
+    ],
+    "debian" => [
+      "buster-20210326",
+      "stretch-20210326",
+      "jessie-20210326"
+    ]
+  }
+
+  def get_os_versions do
+    Map.put(@static_os_versions, "alpine", get_alpine_versions())
+  end
+
+  defp get_alpine_versions() do
+    "library/alpine"
+    |> Bob.DockerHub.fetch_repo_tags()
+    |> Enum.map(fn {version, _} -> Version.parse(version) end)
+    |> Enum.filter(fn
+      {:ok, _version} -> true
+      :error -> false
+    end)
+    |> Enum.map(fn {_valid?, version} -> version end)
+    |> Enum.reduce(%{}, fn version, acc ->
+      Map.update(acc, "#{version.major}.#{version.minor}", version, fn current ->
+        higher_version(current, version)
+      end)
+    end)
+    |> Map.values()
+    |> Enum.sort(:desc)
+    |> Enum.take(@max_alpine_versions)
+    |> Enum.map(fn version -> to_string(version) end)
+  end
+
+  defp higher_version(nil, version), do: version
+  defp higher_version(version, nil), do: version
```

Comment:
When can the versions be `nil`?

Reply:
I had thought that the first version was compared against nil, but that isn't the case since the default value isn't passed through the update function. I'll remove it.